Sources/SwiftFusion/Inference/NewJacobianFactor.swift

```diff
@@ -66,11 +66,25 @@ public struct NewJacobianFactor<
   }
 
   public func errorVector_linearComponent(_ x: Variables) -> ErrorVector {
-    return ErrorVector(jacobian.lazy.map { $0.dot(x) })
+    var r = ErrorVector.zero
+    jacobian.withUnsafeBufferPointer { jacRows in
+      r.withUnsafeMutableBufferPointer { rScalars in
+        for i in 0..<ErrorVector.dimension {
+          rScalars[i] = jacRows[i].dot(x)
+        }
```

Comment:
I started to make all kindsa suggestions here about other approaches and then I realized I was confused about what's being computed here.

----

Would it be good (and optimize well) to write this inner loop as a generic algorithm over collections, so
```suggestion
        rScalars.inplaceDot(jacRows)
```
?

Similarly, once done, you might try:

```
r.inplaceDot(jacobian)
``` 

or even

```
jacobian

Reply:
May not be relevant once I bring the oneliner back.